activesupport/lib/active_support/duration.rb

```diff
@@ -113,19 +116,49 @@ def ago(time = ::Time.current)
     def inspect #:nodoc:
       parts.
         reduce(::Hash.new(0)) { |h,(l,r)| h[l] += r; h }.
-        sort_by {|unit,  _ | [:years, :months, :days, :minutes, :seconds].index(unit)}.
+        sort_by {|unit,  _ | [:years, :months, :weeks, :days, :hours, :minutes, :seconds].index(unit)}.
         map     {|unit, val| "#{val} #{val == 1 ? unit.to_s.chop : unit.to_s}"}.
         to_sentence(locale: ::I18n.default_locale)
     end
 
+    # JSON representation should be a string formatted as ISO 8601 Duration, as recommended by Google JSON Style Guide:
+    # https://google-styleguide.googlecode.com/svn/trunk/jsoncstyleguide.xml?showone=Time_Duration_Property_Values#Time_Duration_Property_Values
     def as_json(options = nil) #:nodoc:
-      to_i
+      iso8601(precision: options && options[:precision])
     end
 
     def respond_to_missing?(method, include_private=false) #:nodoc:
       @value.respond_to?(method, include_private)
     end
 
+    # Creates a new Duration from string formatted according to ISO 8601 Duration.
+    #
+    # See http://en.wikipedia.org/wiki/ISO_8601#Durations
+    # This method allows negative parts to be present in pattern.
+    # If invalid string is provided, it will raise +ActiveSupport::Duration::ISO8601Parser::ParsingError+.
+    def self.parse!(iso8601duration)
+      parts = ISO8601Parser.new(iso8601duration).parse!
+      time  = ::Time.current
+      new(time.advance(parts) - time, parts)
```

Comment:
Yes, it affects.

```
2.2.3 :002 > parts = ActiveSupport::Duration::ISO8601Parser.new('-P1Y3M').parse!
 => {:years=>-1, :months=>-3}
2.2.3 :003 > time_local = Time.current
 => 2015-11-04 18:01:53 +0300
2.2.3 :004 > time_utc   = time_local.dup.utc
 => 2015-11-04 15:01:53 UTC
2.2.3 :005 > time_local.advance(parts)
 => 2014-08-04 18:01:53 +0400
2.2.3 :006 > time_utc.advance(parts)
 => 2014-08-04 15:01:53 UTC
2.2.3 :007 > duration_local = ActiveSupport::Duration.new(time_local.advance(parts) - time_local, parts)
 => -1 years and -3 months
2.2.3 :008 > duration_local.value
 => -39488400.0
2.2.3 :009 > duration_utc = ActiveSupport::Duration.new(time_utc.advance(parts) - time_utc, parts)
 => -1 years and -3 months
2.2.3 :010 > duration_utc.value
 => -39484800.0
2.2.3 :011 > time_local.advance(parts) - time_local
 => -39488400.0
2.2.3 :012 > time_utc.advance(parts) - time_utc
 => -39484800.0
2.2.3 :013 > (-1.year - 3.months).value
 => -39333600.0
2.2.3 :015 > time_local + duration_local
 => 2014-08-04 18:01:53 +0400
2.2.3 :016 > time_utc + duration_utc
 => 2014-08-04 15:01:53 UTC
2.2.3 :017 > time_utc + duration_local
 => 2014-08-04 15:01:53 UTC
2.2.3 :018 > time_local + duration_utc
 => 2014-08-04 18:01:53 +0400
```

1 year and 3 months ago in Moscow was summer time in action. So I created both utc and local times, created durations, examined them and done some math with these duration. Despite the fact, that numeric value of seconds passed to `new` in each case is different (and both are different from “hand made” duration), math results are the same.

I think it's better to stick with `Time.current`, because it's better to have information about local time offset than not to have. If someone want to explicitly work with UTC he should set `Time.zone = 'UTC'` in initializer.

Listing of this test:
```ruby
require_relative 'lib/active_support/all'
Time.zone = 'Moscow'
parts = ActiveSupport::Duration::ISO8601Parser.new('-P1Y3M').parse!
time_local = Time.current
time_utc   = time_local.dup.utc
time_local.advance(parts)
time_utc.advance(parts)
duration_local = ActiveSupport::Duration.new(time_local.advance(parts) - time_local, parts)
duration_local.value
duration_utc = ActiveSupport::Duration.new(time_utc.advance(parts) - time_utc, parts)
duration_utc.value
time_local.advance(parts) - time_local
time_utc.advance(parts) - time_utc
(-1.year - 3.months).value
time_local + duration_local
time_utc + duration_utc
time_utc + duration_local
time_local + duration_utc
```